### PR TITLE
Issue #287 by Inlead: Rewrite ding campaign module.

### PR DESCRIPTION
**Pull request by guddo**
http://platform.dandigbib.org/issues/287

### CODE REVIEW

File: ding_campaign.admin.inc [hunk 1]
```diff
@@ -136,13 +136,16 @@ function ding_campaign_add_rule($form, $form_state) {
...
+  $nid = $form_state['build_info']['args'][0]->nid;
 
   if ($c > 0) {
     // Delete existing rules.
     db_delete('ding_campaign_rules')
-      ->condition('cid', $form_state['build_info']['args'][0]->nid)
+      ->condition('cid', $nid)
       ->execute();
 
+    entity_get_controller('node')->resetCache(array($nid));
```
Comment (kasperg) [inline ding_campaign.admin.inc]: I would appreciate a comment describing why it is necessary to clear the cache.

File: ding_campaign.info [hunk 2]
```diff
@@ -1,11 +1,69 @@
...
+version = 7.x-2.1.0+easy
+project = ding_campaign
 dependencies[] = ctools
-dependencies[] = ding_campaign_ctype
+dependencies[] = entityreference
+dependencies[] = features
+dependencies[] = image
+dependencies[] = list
+dependencies[] = media
+dependencies[] = og_ui
```
Comment (kasperg) [inline ding_campaign.info]: og_ui seems like a peculiar dependency here. Why is this needed?

File: ding_campaign.info [hunk 3]
```diff
@@ -1,11 +1,69 @@
...
 description = Campaign system. Displays context-sensitive, ad-like campaigns.
-package = Ding!
-version = "7.x-0.4"
 core = 7.x
-configure = admin/config/ding/campaign
-dependencies[] = field
-dependencies[] = image
-dependencies[] = features
+package = Ding!
+version = 7.x-2.1.0+easy
```
Comment (kasperg) [inline ding_campaign.info]: +easy should not be added to the version.

**Comment by kasperg** [comment]
@guddo: I have reviewed this. In general things look great.

I have a request for an inline comment and a version should be adjusted as well.

File: ding_campaign.info [hunk 4]
```diff
@@ -1,11 +1,69 @@
...
-dependencies[] = field
-dependencies[] = image
-dependencies[] = features
-dependencies[] = ctools
-dependencies[] = ding_campaign_ctype
+name = Ding campaign
+description = Campaign system. Displays context-sensitive, ad-like campaigns.
+core = 7.x
+package = Ding!
+version = 7.x-2.3.1+easy
```
Comment (kasperg) [inline ding_campaign.info]: Please remove versioning information from the module.

Comment (guddo) [inline ding_campaign.info]: Sorry, this is a PR directly from our distribution - feel free to fork our repo or duplicate a PR like this.

**Comment by kasperg** [comment]
Merged. Thanks.

**Comment by guddo** [comment]
@kasperg: See https://github.com/ding2/ding_campaign/pull/18.
